examples/internal/integration/integration_test.go

```diff
@@ -2504,3 +2505,41 @@ func testABETrace(t *testing.T, port int) {
 		return
 	}
 }
+
+func testEchoWithNonASCIIHeaders(t *testing.T, port int, apiPrefix string) {
+	apiURL := fmt.Sprintf("http://localhost:%d/%s/example/echo/myid", port, apiPrefix)
+
+	req, err := http.NewRequest("POST", apiURL, strings.NewReader("{}"))
+	if err != nil {
+		t.Errorf("http.NewRequest() = err: %v", err)
+		return
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Grpc-Metadata-Location", "Gjøvik")
```

Comment:
Can we also test an invalid header key?

Reply:
Done.

Note that the same kind of invalidness wouldn't work, because "Grpc-Metadata-Gjøvik" is actually an invalid header name on the HTTP level, and it gets rejected by net/http. So some of these checks are redundant.

Turns out there are still characters that are allowed in HTTP header names but not gRPC metadata keys; "*^'&%|!#$`~+". So testing with the header name `Grpc-Metadata-Foo+Bar`.